#! /bin/sh /usr/share/dpatch/dpatch-run
## 30_ocnst_char.dpatch by Paul Brossier <piem@debian.org>
##
## DP: Avoid const char to char conversion

@DPATCH@
diff --git a/input/flac/FlacPlugin.cpp b/input/flac/FlacPlugin.cpp
index 8abbf43..9eb0640 100644
--- a/input/flac/FlacPlugin.cpp
+++ b/input/flac/FlacPlugin.cpp
@@ -239,7 +239,7 @@ flac_stream_info (input_object * obj, stream_info * info)
     else
     {
 	// use stream name
-	char * fname = strrchr (f->name ().c_str (), '/');
+	const char * fname = strrchr (f->name ().c_str (), '/');
 	if (fname)
 	{
 	    fname++;
@@ -270,7 +270,7 @@ flac_can_handle (const char * name)
 	if (strncmp(name, "http://", 7) == 0) {
 		return 0.0;
 	}
-	char *ext = strrchr(name, '.');
+	const char *ext = strrchr(name, '.');
 	if (!ext)
 		return 0.0;
 	ext++;
@@ -327,9 +327,9 @@ input_plugin_info (void)
 
     flac_plugin.version      = INPUT_PLUGIN_VERSION;
 #ifdef LEGACY_FLAC
-    flac_plugin.name         = "flac player v1.1.2";
+    flac_plugin.name         = (char *)"flac player v1.1.2";
 #else
-    flac_plugin.name         = "flac player v1.1.3/1.1.4/1.2";
+    flac_plugin.name         = (char *)"flac player v1.1.3/1.1.4/1.2";
 #endif
     flac_plugin.author       = "Drew Hess";
     flac_plugin.init         = flac_init;
